### PR TITLE
Pre release v1.7

### PR DESCRIPTION
**Pull request by hicolour**
- [x] java8 introduction - improper usage of map interface contract (minor core changes  & tests)
- [x] java7/java8 compilation / travisci config
- [x] minor grovy upgrade -> 2.4.4

### CODE REVIEW

File: smooks-core/src/main/java/org/milyn/payload/JavaResult.java [hunk 1]
```diff
@@ -42,7 +42,7 @@
      * Public default constructor.
      */
     public JavaResult() {
-        this(false);
+        this(true);
```
Comment (tfennelly) [inline smooks-core/src/main/java/org/milyn/payload/JavaResult.java]: Why this change? Might it cause compatibility issues for people already depending on this behaviour?

Comment (hicolour) [inline smooks-core/src/main/java/org/milyn/payload/JavaResult.java]: on the contrary, it's more or less to give them some backward compatibility (previously hashmap behaves simillar to linkedhashmap )- this is not necessary and if you want I can bring back previous behaviour

Comment (tfennelly) [inline smooks-core/src/main/java/org/milyn/payload/JavaResult.java]: Okay ... I'll take your word for it (and blame you if others run into issues because of it 😉  )
